miner/src/block_assembler.rs

```diff
@@ -135,31 +132,41 @@ impl<CS: ChainStore + 'static> BlockAssembler<CS> {
 
         let new_uncle_receiver = notify.subscribe_new_uncle(BLOCK_ASSEMBLER_SUBSCRIBER);
         let thread = thread_builder
-            .spawn(move || loop {
-                select! {
-                    recv(signal_receiver) -> _ => {
-                        break;
-                    }
-                    recv(new_uncle_receiver) -> msg => match msg {
-                        Ok(uncle_block) => {
-                            let hash = uncle_block.header().hash();
-                            self.candidate_uncles.insert(hash.to_owned(), uncle_block);
-                            self.last_uncles_updated_at
-                                .store(unix_time_as_millis(), Ordering::SeqCst);
-                        }
-                        _ => {
-                            log::error!(target: "miner", "new_uncle_receiver closed");
+            .spawn(move || {
+                let mut candidate_uncles = LruCache::new(MAX_CANDIDATE_UNCLES);
```

Comment:
Why move from the field to a local variable? Work around the conflict mutable borrow?

Reply:
`Move out `,   `RefCel `l， `Mutex `
I choose  `Move out`